skylib/hash.bzl

```diff
@@ -0,0 +1,23 @@
+"""Functions for producing the hash of an artifact."""
```

Comment:
could you add a comment here pointing to the upstream hash.bzl in bazel_tools and explain why we forked it? that way we have a chance of comparing this against future changes to that one.